pkg/services/helpers/neofs/neofs.go

```diff
@@ -244,31 +245,31 @@ func parseRange(s string) (*object.Range, error) {
 	return r, nil
 }
 
-// ObjectSearch returns a list of object IDs from the provided container.
-func ObjectSearch(ctx context.Context, c Client, priv *keys.PrivateKey, containerIDStr string, prm client.PrmObjectSearch) ([]oid.ID, error) {
+// ObjectSearch returns a list of object IDs with attributes from the provided container.
+func ObjectSearch(ctx context.Context, c Client, priv *keys.PrivateKey, containerIDStr string, filters object.SearchFilters, attrs []string) ([]client.SearchResultItem, error) {
 	var (
 		s           = user.NewAutoIDSignerRFC6979(priv.PrivateKey)
-		objectIDs   []oid.ID
 		containerID cid.ID
+		allResults  []client.SearchResultItem
+		cursor      string
 	)
 	err := containerID.DecodeString(containerIDStr)
 	if err != nil {
 		return nil, fmt.Errorf("%w: %w", ErrInvalidContainer, err)
 	}
-	reader, err := c.ObjectSearchInit(ctx, containerID, s, prm)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initiate object search: %w", err)
-	}
-	defer reader.Close()
 
-	err = reader.Iterate(func(oid oid.ID) bool {
-		objectIDs = append(objectIDs, oid)
-		return false
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error during object IDs iteration: %w", err)
+	for {
+		results, nextCursor, err := c.SearchObjects(ctx, containerID, filters, attrs, cursor, s, client.SearchObjectsOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to search objects: %w", err)
+		}
+		allResults = append(allResults, results...)
+		if nextCursor == "" {
+			break
+		}
+		cursor = nextCursor
 	}
```

Comment:
Move this loop to a separate goroutine, send results to the resulting channel. Close channel when all results are retrieved by SEARCH. You may see as an example: https://github.com/nspcc-dev/neo-go/blob/92523bb7d529f41ee052643cf614ea99e6c31c99/pkg/core/storage/memcached_store.go#L171